src/sql/workbench/services/languageAssociation/common/doHandleUpgrade.ts

```diff
@@ -20,8 +20,8 @@ export function doHandleUpgrade(editor?: EditorInput): EditorInput | undefined {
 			editor.getPreferredMode();
 		}
 		const association = languageRegistry.getAssociationForLanguage(language);
-		if (association) {
-			return association.convertInput(editor);
+		if (association && association.syncConvertinput) {
```

Comment:
There's other places we call convertInput - you aren't going to update those?

Reply:
this is the only one that can't handle a promise. the other places we call convert input I've updated to handle promise.